### DISCUSSION TITLE
Ability to set `namespace` as prefix

### DISCUSSION THREAD

**Discussion by ViacheslavKudinov** [PaulHatch/semantic-version#129] [Ideas]
Hello,
currently, `namespace` always will be added as a suffix, eg `v1.0.1-namespace`.
It would be good if we could set where to put `namespace` via `version_format` input. 
Maybe something like:
`version_format: "${namespace}-${major}.${minor}.${patch}"`, with default as it is: `version_format: "${major}.${minor}.${patch}-${namespace}"`.

Maybe it can be another suggestion how to set it as prefix.

Thank you in advance.

**Comment by PaulHatch** [comment]
Actually I think you could actually just set the prefix and it will act as a "namespace", with some possible edge cases if your prefix includes numbers or one prefix is a substring of another. Prefix was not intended for such use but I believe it would work, the namespace unit tests all pass when I change them to use `prefix` rather than `namespace`.

The `version_format` parameter is used to output the version and has nothing to do with the tag format, there are two distinct types of "versions", the tag version and the actual app/service version. I somewhat regret adding this because it's been the source of some confusion on exactly this point. Adding a corresponding format string for tags is not really practical since this value is parsed in addition to being used to format the output, and as well as being complicated to implement would have similar edge cases described above.

**Comment by ViacheslavKudinov** [comment]
Hi,
thank you for the answer.
I got that `version_format` doesn't affect tag anyhow, and it was wrong suggestion from my side.
We use in our company versioning when we have prefix (usually just a string, like Terrafrom module name from monorepo).
Currently, we can get with `namespace` input tags like "v1.0.0-module-B", but i would like to get tags like "module-B-v1.0.0" if possible.
When you mentioned i can use "prefix" now, what do you mean by that? 
I do not see the input for it, did I miss something?
Thank you in advance.

**Comment by ViacheslavKudinov** [comment]
Was it this one https://github.com/PaulHatch/semantic-version/blob/master/action.yml#L11 ?

Eg to set `tag_prefix `input like  "module-B-v" and so on ?

**Reply by PaulHatch** [reply]
Yes. It will not be recognized as a valid tag unless the prefix matches, which is what the namespace is doing also.

**Reply by ViacheslavKudinov** [reply]
OK, thanks.
I will give it a try if it works as I wanted.

**Reply by ViacheslavKudinov** [reply]
It seems it works as I wanted.
Thanks @PaulHatch  a lot for the tip how it can be achieved by `tag_prefix` input 👍

I will close the discussion.
Thank you for the help!
